### PR TITLE
Dev/shgu/testing

### PR DESCRIPTION
**Pull request by Shyam-Gupta**
Added guidance for:
1. Unit tests for Public and Internal methods
2. Code Coverage

### CODE REVIEW

File: Documentation/testing.md [hunk 1]
```diff
@@ -71,7 +71,10 @@ Naming
 
 Strategy
 * **Unit tests should be part of the same PR as code changes**
-  * Please test internal methods as well, not just publics. 
+  * Unit tests must be added for any change to Public APIs. We will accept unit tests for internal methods as well.
```
Comment (Tanya-Solyanik) [inline Documentation/testing.md]: I'm not sure if Public is usually capitalized in the docs...

File: Documentation/testing.md [hunk 2]
```diff
@@ -71,7 +71,10 @@ Naming
 
 Strategy
 * **Unit tests should be part of the same PR as code changes**
```
Comment (Tanya-Solyanik) [inline Documentation/testing.md]: @zsd4yr @Shyam-Gupta do you guys know how common term `PR` is... Should we spell it out?

File: Documentation/testing.md (hunk 1 above)
Comment (Shyam-Gupta) [reply Documentation/testing.md]: ok. I will change it to public.

File: Documentation/testing.md (hunk 2 above)
Comment (Shyam-Gupta) [reply Documentation/testing.md]: I think its best to avoid abbreviations. I will change it to Pull Request.

Comment (karelz) [reply Documentation/testing.md]: Nit: PR is well established term on GitHub. We have it in our docs already. It's almost weird to see it spelled out :)

File: Documentation/testing.md [hunk 3]
```diff
@@ -70,8 +70,11 @@ Naming
   * This is very useful when viewing test results, and when browsing in the test explorer
 
 Strategy
-* **Unit tests should be part of the same PR as code changes**
-  * Please test internal methods as well, not just publics. 
+* **Unit tests should be part of the same Pull request as code changes**
+  * Unit tests must be added for any change to public APIs. We will accept unit tests for internal methods as well. 
+* **Code Coverage**
+  * In Visual Studio Test Explorer, select all tests, right click and execute 'Analyze code coverage for selected tests' command. This will run all tests and give a summary of blocks covered in 'Code Coverage Results' window. The summary can be drilled down to method level.   
+  Any code change accompanied with unit tests is expected to increase code coverage for the code modified.
```
Comment (zsd4yr) [inline Documentation/testing.md]:
```suggestion
  * Any code change accompanied with unit tests is expected to increase code coverage for the code modified. 
```

**Pull request merged**